packages/server/src/fhircast/websocket.ts

```diff
@@ -1,11 +1,56 @@
 import { generateId } from '@medplum/core';
 import { AsyncLocalStorage } from 'node:async_hooks';
 import { IncomingMessage } from 'node:http';
+import os from 'node:os';
 import ws from 'ws';
 import { DEFAULT_HEARTBEAT_MS, heartbeat } from '../heartbeat';
 import { globalLogger } from '../logger';
+import { setGauge } from '../otel/otel';
 import { getRedis, getRedisSubscriber } from '../redis';
 
+const hostname = os.hostname();
+const METRIC_OPTIONS = { attributes: { hostname } };
+let heartbeatHandler: (() => void) | undefined;
+
+const websocketMap = new Map<ws.WebSocket, string>();
+const topicRefCountMap = new Map<string, number>();
+
+function initHeartbeatHandler(): void {
+  if (!heartbeatHandler) {
+    heartbeatHandler = (): void => {
+      const baseHeartbeatPayload = {
+        timestamp: new Date().toISOString(),
+        id: generateId(),
+        event: {
+          context: [{ key: 'period', decimal: `${Math.ceil(DEFAULT_HEARTBEAT_MS / 1000)}` }],
+          'hub.event': 'heartbeat',
+        },
+      };
+
+      const redis = getRedis();
+      for (const projectAndTopic of topicRefCountMap.keys()) {
+        redis
+          .publish(
+            projectAndTopic as string,
+            JSON.stringify({
+              ...baseHeartbeatPayload,
+              event: { ...baseHeartbeatPayload.event, 'hub.topic': projectAndTopic.split(':')[1] },
+            })
+          )
+          .catch(console.error);
+      }
+
+      setGauge('medplum.fhircast.websocketCount', websocketMap.size, METRIC_OPTIONS);
```

Comment:
New metrics for FHIRcast `websocketCount` and `topicCount`, just for future reference